beacon_node/network/src/sync/manager.rs

```diff
@@ -567,48 +688,119 @@ impl<T: BeaconChainTypes> SyncManager<T> {
                 beacon_block,
                 seen_timestamp,
             } => {
-                self.rpc_block_or_blob_received(
-                    request_id,
-                    peer_id,
-                    beacon_block.into(),
-                    seen_timestamp,
-                );
+                self.rpc_block_received(request_id, peer_id, beacon_block, seen_timestamp);
             }
+            SyncMessage::RpcBlob {
+                request_id,
+                peer_id,
+                blob_sidecar,
+                seen_timestamp,
+            } => self.rpc_blob_received(request_id, peer_id, blob_sidecar, seen_timestamp),
             SyncMessage::UnknownBlock(peer_id, block, block_root) => {
-                // If we are not synced or within SLOT_IMPORT_TOLERANCE of the block, ignore
-                if !self.network_globals.sync_state.read().is_synced() {
-                    let head_slot = self.chain.canonical_head.cached_head().head_slot();
-                    let unknown_block_slot = block.slot();
-
-                    // if the block is far in the future, ignore it. If its within the slot tolerance of
-                    // our current head, regardless of the syncing state, fetch it.
-                    if (head_slot >= unknown_block_slot
-                        && head_slot.sub(unknown_block_slot).as_usize() > SLOT_IMPORT_TOLERANCE)
-                        || (head_slot < unknown_block_slot
-                            && unknown_block_slot.sub(head_slot).as_usize() > SLOT_IMPORT_TOLERANCE)
-                    {
-                        return;
+                let block_slot = block.slot();
+
+                if self.synced_and_connected_within_tolerance(block_slot, &peer_id) {
+                    let parent_root = block.parent_root();
+                    self.block_lookups.search_parent(
+                        block_slot,
+                        block_root,
+                        parent_root,
+                        peer_id,
+                        &mut self.network,
+                    );
+                    if self.should_delay_lookup(block_slot) {
```

Comment:
the intent here's to immediately make the parent request and to delay the request for any missing blobs for the child block

Reply:
Just to check my understanding, the reason we want to delay the lookup is because immediately making the lookup will lead to multiple other requests being spawned. For e.g. if we get blob1 and immediately make a request for all other blobs + block, then getting blob2 over rpc will spawn requests for blob3 + blob4 again even though they are probably en route.
Would be nice to document why exactly we are delaying here